engine/runtime/src/test/java/org/enso/interpreter/node/expression/builtin/meta/TypeOfNodeTest.java

```diff
@@ -0,0 +1,90 @@
+package org.enso.interpreter.node.expression.builtin.meta;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+import com.oracle.truffle.api.interop.TruffleObject;
+import java.util.concurrent.Callable;
+import org.enso.interpreter.runtime.callable.UnresolvedSymbol;
+import org.enso.interpreter.runtime.error.DataflowError;
+import org.graalvm.polyglot.Context;
+import org.graalvm.polyglot.Value;
+import org.graalvm.polyglot.proxy.ProxyExecutable;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+public class TypeOfNodeTest {
+
+  private static Context ctx;
+
+  @BeforeClass
+  public static void initCtx() throws Exception {
+    ctx = Context.create();
+  }
+
+  @AfterClass
+  public static void disposeCtx() throws Exception {
+    ctx.close();
+  }
+
+  @Test
+  public void typeOfUnknownSymbol() throws Exception {
+    assertType("Function", false);
+  }
+
+  @Test
```

Comment:
I think it may be beneficial to add a link to the PR / issue description, as without the whole context of this bug, this test may come as _very very_ specific and unexpected.

Reply:
This is a regular test for testing specialization (at least for me). PR number is available in `git blame`. Details, discussions (including your comment) is going to be available in the PR.